docs/user/installation.rst

```diff
@@ -4,50 +4,239 @@
+
+.. code-block:: bash
+
+  pip install --upgrade https://github.com/Theano/Theano/archive/master.zip
+  pip install --upgrade https://github.com/Lasagne/Lasagne/archive/master.zip
+
+Again, add ``--user`` if you want to install to your home directory instead.
+
+
+Development installation
+========================
+
+Alternatively, you can install Lasagne (and optionally Theano) from source,
+in a way that any changes to your local copy of the source tree take effect
+without requiring a reinstall. This is often referred to as *editable* or
+*development* mode. Firstly, you will need to obtain a copy of the source tree:
 
 .. code-block:: bash
 
   git clone https://github.com/Lasagne/Lasagne.git
 
-Once you have obtained a copy of the source, run the following command to
-install Lasagne's dependencies:
+It will be cloned to a subdirectory called ``Lasagne``. Make sure to place it
+in some permanent location, as for an *editable* installation, Python will
+import the module directly from this directory and not copy over the files.
+Enter the directory and install the known good version of Theano:
 
 .. code-block:: bash
 
   cd Lasagne
   pip install -r requirements.txt
 
-To install the Lasagne package itself, run:
+Alternatively, install the bleeding-edge version of Theano as described in the
+previous section.
+
+To install the Lasagne package itself, in editable mode, run:
+
+.. code-block:: bash
+
+  pip install . --editable
+
+As always, add ``--user`` to install it to your home directory instead.
+
+**Optional**: If you plan to contribute to Lasagne, you will need to fork the
+Lasagne repository on GitHub. This will create a repository under your user
+account. Update your local clone to refer to the official repository as
+``upstream``, and your personal fork as ``origin``:
+
+.. code-block:: bash
+
+  git remote rename origin upstream
+  git remote add origin https://github.com/<your-github-name>/Lasagne.git
+
+If you set up an `SSH key <https://help.github.com/categories/ssh/>`_, use the
+SSH clone URL instead: ``git@github.com:<your-github-name>/Lasagne.git``.
+
+You can now use this installation to develop features and send us pull requests
+on GitHub, see :doc:`development`!
+
+  
+Installation without pip
+========================
+
+If, for some reason, you need to install Theano and Lasagne without ``pip``,
+you will need to obtain the source trees:
+
+.. code-block:: bash
+
+  git clone https://github.com/Theano/Theano.git
+  git clone https://github.com/Lasagne/Lasagne.git
+
+And then install via ``setup.py``:
 
 .. code-block:: bash
 
-  python setup.py install  # or 'setup.py develop' if you're developing Lasagne itself
+  cd Theano && python setup.py install && cd -
+  cd Lasagne && python setup.py install && cd -
+
+Append ``--prefix=~/.local`` after ``setup.py`` to install to your home
+directory, and replace ``install`` with ``develop`` to install in editable
```

Comment:
I would reorder the parts of this sentence a bit, starting with "To install in editable mode, append...". Otherwise it's a bit unclear that these instructions are specific to that use case.